src/gateway/routers/Aggregator.sol

```diff
@@ -189,15 +207,53 @@ contract Aggregator is Auth, IAggregator {
 
     // --- Outgoing ---
     /// @inheritdoc IAggregator
-    function send(bytes calldata message) external auth {
+    function send(bytes calldata message) external payable auth {
         uint256 numRouters = routers.length;
         require(numRouters > 0, "Aggregator/not-initialized");
 
+        uint256 fuel = msg.value;
+        uint256 centrifugeCost;
+
+        if (fuel > 0) {
+            centrifugeCost = centrifugeGasService.estimate(message);
+        }
+
         bytes memory proof = abi.encodePacked(uint8(MessagesLib.Call.MessageProof), keccak256(message));
         for (uint256 i; i < numRouters; i++) {
-            RouterLike(routers[i]).send(i == PRIMARY_ROUTER_ID - 1 ? message : proof);
+            RouterLike currentRouter = RouterLike(routers[i]);
+            bytes memory payload = i == PRIMARY_ROUTER_ID - 1 ? message : proof;
+            // TODO a discussion:
+            // Assumption 1: We would like to be able to call `send` without providing any funds for gas payments
+            // Assumption 2: Funds for gas payments might be provided by EOA  but the estimated value could be retrieved
+            // on a previous block.
+            // Assumption 3: There is volatility and fluctuation in axelar Cost and centrifuge Cost
+            //
+            // Resolution 1: If no one is paying for the gas funds, there is no point in providing an array of
+            // gas costs for each router from the outside
+            // Resolution 2: During the TX runtime, we get latest updated value in all our contract for all the fee
+            // and make calculation based on that. If the provide value "fuel" turns out to be not enough, we receive
+            // a faster feedback and we don't have to submit the tx on axelar or any other bridge and wait for a
+            // feedback that
+            // the tx will end up underpaid
+            // Resolution 3: Resolution 2 applies - we get the latest updated costs on centra and axelar assuming they
+            // got updated
+
+            if (fuel > 0) {
+                uint256 txCost = currentRouter.estimate(payload) + centrifugeCost;
```

Comment:
I would pass `centrifugeCost` to `estimate` and include it in the calculation there. There might be an adapter in the future that actually takes a % of the total cost rather than a fixed fee, and this would then allow implementing that (so the cost of the adapter would be `1.1 * centrifugeCost` as an example)

Reply:
Just for the protocol: I don't feel like passing centrifuge cost  to the estimate method is the right way. IMO the `GasService` should be a dependency  in every adapter.  It is an essential part of it. If we use `adapter.estimate` in other contracts we have to inject `GasService` there as well only to get the `centrifugeCost`  and pass it to the `estimate` method.
The only reason why I also agree with this approach is the fact that if we put `GasService` as a dep on each router, there will be calls to the gas service as many times as the amount of routers instead of having it just once in the Aggregator.
I will make the change I just wanted to share my though.